sw/device/tests/dif_gpio/dif_gpio_test.cc

```diff
@@ -0,0 +1,85 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "mock_mmio.h"
+
+extern "C" {
+#include "gpio_regs.h"
+#include "sw/device/lib/dif/dif_gpio.h"
+}
+
+// This is a temporary measure to work around the dependency on
+// |GPIO0_BASE_ADDR| in |GPIO_*()| family of macros in gpio_regs.h.
+// See lowrisc/opentitan#1118.
+static const uintptr_t GPIO0_BASE_ADDR = 0x0U;
+#define DEFINE_GPIO_REG_CONSTANT(NAME) \
+  static const ptrdiff_t GPIO_##NAME##_REG_OFFSET = GPIO_##NAME(0);
+DEFINE_GPIO_REG_CONSTANT(INTR_STATE)
+DEFINE_GPIO_REG_CONSTANT(INTR_ENABLE)
+DEFINE_GPIO_REG_CONSTANT(INTR_TEST)
+DEFINE_GPIO_REG_CONSTANT(DATA_IN)
+DEFINE_GPIO_REG_CONSTANT(DIRECT_OUT)
+DEFINE_GPIO_REG_CONSTANT(MASKED_OUT_LOWER)
+DEFINE_GPIO_REG_CONSTANT(MASKED_OUT_UPPER)
+DEFINE_GPIO_REG_CONSTANT(DIRECT_OE)
+DEFINE_GPIO_REG_CONSTANT(MASKED_OE_LOWER)
+DEFINE_GPIO_REG_CONSTANT(MASKED_OE_UPPER)
+DEFINE_GPIO_REG_CONSTANT(INTR_CTRL_EN_RISING)
+DEFINE_GPIO_REG_CONSTANT(INTR_CTRL_EN_FALLING)
+DEFINE_GPIO_REG_CONSTANT(INTR_CTRL_EN_LVLHIGH)
+DEFINE_GPIO_REG_CONSTANT(INTR_CTRL_EN_LVLLOW)
+DEFINE_GPIO_REG_CONSTANT(CTRL_EN_INPUT_FILTER)
+#undef DEFINE_GPIO_REG_CONSTANT
+
+using ::testing::Return;
+
+// Equal to operator to be able to write EXPECT_EQ(expGpio, actGpio).
```

Comment:
I am... skeptical that an ADL operator is an acceptable cost for making `EXPECT_EQ` work. Consider defining your own `EXPECT_THAT` matcher instead?